hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -131,7 +131,8 @@ public ClientConnectionManagerImpl(HazelcastClientInstanceImpl client, AddressTr
         ClientNetworkConfig networkConfig = config.getNetworkConfig();
 
         final int connTimeout = networkConfig.getConnectionTimeout();
-        this.connectionTimeout = connTimeout == 0 ? Integer.MAX_VALUE : connTimeout;
+        final int attemptPeriod = networkConfig.getConnectionAttemptPeriod();
+        this.connectionTimeout = Math.min(connTimeout, attemptPeriod);
```

Comment:
Ok, reverted this change. Modified the test and added the connection timeout config to the tests client config.